src/services/notification/telegram.rs

```diff
@@ -211,33 +222,31 @@ impl Notifier for TelegramNotifier {
 	/// # Returns
 	/// * `Result<(), NotificationError>` - Success or error
 	async fn notify(&self, message: &str) -> Result<(), NotificationError> {
-		// Add message and disable_web_preview to URL parameters
-		let mut url_params = self.inner.url_params.clone().unwrap_or_default();
-		url_params.insert("text".to_string(), message.to_string());
-		url_params.insert(
+		// Get default payload fields
+		let mut payload_fields = self.inner.payload_fields.clone().unwrap_or_default();
+
+		// Add the dynamic fields for this specific notification.
+		payload_fields.insert("text".to_string(), serde_json::json!(message));
+		payload_fields.insert(
 			"disable_web_page_preview".to_string(),
-			self.disable_web_preview.to_string(),
+			serde_json::json!(self.disable_web_preview),
 		);
 
-		// Create a new WebhookNotifier with updated URL parameters
-		let notifier = WebhookNotifier::new(WebhookConfig {
-			url: self.inner.url.clone(),
-			url_params: Some(url_params),
-			title: self.inner.title.clone(),
-			body_template: self.inner.body_template.clone(),
-			method: Some("GET".to_string()),
-			secret: None,
-			headers: self.inner.headers.clone(),
-			payload_fields: None,
-		})?;
-
-		notifier.notify_with_payload(message, HashMap::new()).await
+		// Send the notification using the inner Webhook notifier
+		self.inner
+			// TODO: The `message` parameter is required by the Notifier trait for generic 
+			// webhook signing, but it's duplicated in this specific payload
+			.notify_with_payload(message, payload_fields)
```

Comment:
I suppose because telegram has the message in payload fields already?

Just checking, but did you test if POST with these fields work? I wonder why we were doing a GET with url params before..